### PR TITLE
Properly finalize statements when closing connections

### PR DESCRIPTION
**Pull request by FiloSottile**
If close_v2 is called on a connection with unfinalized statements it
enters a zombie status, waiting for the statements to be finalized. This
leaks resources and prevents the WAL file from being deleted (the latter
being how I found this out).

close_v2 is for languages with unordered but reliable finalizers anyway.

### CODE REVIEW

File: sqlite.go [hunk 1]
```diff
@@ -178,12 +178,15 @@ func openConn(path string, flags OpenFlags) (*Conn, error) {
...
-// Close closes the database connection using sqlite3_close_v2.
-// https://www.sqlite.org/c3ref/close.html
+// Close closes the database connection using sqlite3_close and finalizes
+// persistent prepared statements. https://www.sqlite.org/c3ref/close.html
 func (conn *Conn) Close() error {
 	conn.cancelInterrupt()
 	conn.closed = true
-	res := C.sqlite3_close_v2(conn.conn)
+	for _, stmt := range conn.stmts {
+		stmt.Finalize()
```
Comment (crawshaw) [inline sqlite.go]: What do you think about returning an error if a statement is in the middle of returning results? We could check `stmt.lastHasRow` here.

For a later commit.

**Comment by crawshaw** [comment]
Thanks!

File: sqlite.go (hunk 1 above)
Comment (FiloSottile) [reply sqlite.go]: I think it's not unexpected for an application to decide half-way through some results that it got all it needs or it doesn't like what it sees and decide to close the connection.

Comment (crawshaw) [reply sqlite.go]: That's fair (I don't consider the single-thread case enough). I think what I want is a check for open results on `Pool.Close`.

**Pull request merged**